Documentation/proposals/202305-scrapeclasses.md

```diff
@@ -0,0 +1,189 @@
+Allow the user to select a scrape class for the probe.
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Probe
+spec:
+  scrapeClass: istio-mtls
+```
+
+### ServiceMonitor Resource
+Allow the user to select a scrape class for each endpoint.
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+spec:
+  endpoints:
+  - port: http
+    path: /metrics
+    scrapeClass: istio-mtls
+```
+
+Deprecate the unsafe TLS settings in `ServiceMonitor`.
+
+## Test Plan
+
+1. Regression test; ensure no change to generated configs unless a scrape class is applied.
+2. Optionality of the default scrape class; ensure that it is an optional configuration element.
+3. Acceptance testing in Istio environment; ensure that the solution is effective for a key use case.
+
+## Alternatives
+
+### Global scrape TLS configuration
+An alternative solution would be to apply a default TLS configuration to all monitors.
+
+For example, via a hypothetical field `spec.scrapeTlsConfig`:
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+spec:
+  scrapeTlsConfig:
+    caFile: "/etc/istio-certs/root-cert.pem"
+    certFile: "/etc/istio-certs/cert-chain.pem"
+    keyFile: "/etc/istio-certs/key.pem"
+    insecureSkipVerify: true
+```
+
+Objections:
+1. A singular default configuration may be too inflexible to effectively scrape a diverse set of pods. For example, in Istio, 
+some pods may be in STRICT mode.
+
+### Istio Permissive Mode
+An alternative for the Istio use case is to use PERMISSIVE mode (see [documentation](https://istio.io/latest/docs/concepts/security/#permissive-mode)),
+or to use `exclude` annotations on the pod such that the metrics endpoint bypasses mTLS.
+
+Objections:
+1. This solution is less secure at a transport level unless TLS is implemented at the application layer.
+2. Strict mTLS is the basis for [Istio authorization policy](https://istio.io/latest/docs/reference/config/security/authorization-policy/), e.g. to explicitly allow access from prom to the metrics endpoint.
+
+### ScrapeClass Resource
+A variant of the proposed solution is to introduce a new custom resource for defining scrape classes.
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ScrapeClass
+metadata:
+  name: istio
+spec:
+  tlsConfig:
+    caFile: "/etc/istio-certs/root-cert.pem"
+    certFile: "/etc/istio-certs/cert-chain.pem"
+    keyFile: "/etc/istio-certs/key.pem"
+    insecureSkipVerify: true
+```
+
+An open question is whether the resource would be cluster-scoped or namespace-scoped.
+
+Objections:
+1. Since the file paths are dependent on the volume mounts in the server, this approach may not achieve a meaningful decoupling.
+
+### Non-Safe Monitors
+Another alternative would be to allow `PodMonitor` and `Probe` to use unsafe TLS settings. 
+
+Objections:
+1. See [explanation](https://github.com/prometheus-operator/prometheus-operator/issues/3922#issuecomment-802899950) for why 
+unsafe settings were disallowed in the first place.
```

Comment:
As far as I understand, there isn't a real security concern for TLS parameters because the target never receives the TLS key in clear. But the issue exists for tokens, basic auth, ...

Anyway the current approach used for ServiceMonitor isn't something that we want to generalize because requiring that the owner of the ServiceMonitor knows in advance the location of the client TLS key & cert files in the Prometheus container creates too much coupling.